### ISSUE TITLE
google_sign_in requires too much (unnecessary?) setup work.

### ISSUE THREAD

**Issue by james-lawrence** [flutter/flutter#25908]
I'm attempting to pull a profile avatar, and google sign in seems to be the closest I can get to that in flutter. so if this is the wrong place forgive me.

however it seems to require google-services.json file, which seems to come from firebase.
I don't want to have to manage yet another system.

google play services make no references to firebase in its [setup documentation](https://developers.google.com/android/guides/setup).

the plugin itself has no imports of firebase code.... all of this leads me to believe this is unnecessary for the actual signin functionality and there is a simpler way to setup the plugin without firebase.

there is also a TODO:
```dart
        // Only requests a clientId if google-services.json was present and parsed
        // by the google-services Gradle script.
        // TODO(jackson): Perhaps we should provide a mechanism to override this
        // behavior.
        int clientIdIdentifier =
            registrar
                .context()
                .getResources()
                .getIdentifier(
                    "default_web_client_id", "string", registrar.context().getPackageName());
        if (clientIdIdentifier != 0) {
```
tl;dr; is firebase really necessary for simple signin on android?

**Comment by james-lawrence** [comment]
side note: what lead me down this path is I'm seeing an error in the logs when attempting to use this plugin:

```12-30 10:33:47.606   320   320 W SignIn  : [Activity,SignInChimeraActivity] Failed to record the consent.```

with no additional information, and further digging into the relevant APIs only thing I can guess is that I am missing an android oauth2 client ID.

however manually specifying the client ID in the code did nothing to resolve the error.

**Comment by james-lawrence** [comment]
I'm happy to implement this feature once #25909 is addressed.

**Comment by ericmand** [comment]
@james-lawrence do the comments in #36673 explaining on how to implement Google sign in with just a few steps and no use of Firebase answer your question?

**Comment by iapicca** [comment]
Without additional information, we are unfortunately not sure how to resolve this issue. We are therefore reluctantly going to close this bug for now. Please don't hesitate to comment on the bug if you have any more information for us; we will reopen it right away!  
Thanks for your contribution.

Could everyone who still has this problem please file a new issue with the exact descriptions what happens, logs and the output of 'flutter doctor -v'  please.
All system setups can be slightly different so it's always better to open new issues and reference related issues.
